ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1302,7 +1302,7 @@ void checkAndMaybeComplete() {
         // the chunk is complete now. We can charge against quota for the chunk.
         if (quotaChargeCallback != null) {
           try {
-            quotaChargeCallback.chargeQuota();
+            quotaChargeCallback.chargeQuota(chunkBlobProperties.getBlobSize());
```

Comment:
we should check if this is a metadata put chunk and the exception if so we don't have to charge quota.
```
if (quotaChargeCallback != null && chunkException == null && !isMetadataChunk()) {
    ....
}
```